src/codeql.ts

```diff
@@ -76,51 +77,97 @@ export async function runQuery(
     queryPackName,
   ]);
 
-  const bqrsFilePath = path.join("results", "results.bqrs");
-  const tempBqrsFilePath = getBqrsFile(databaseName);
-  fs.renameSync(tempBqrsFilePath, bqrsFilePath);
+  const queryPaths = await getRemoteQueryPackQueries(codeql, queryPack);
 
-  const bqrsInfo = await getBqrsInfo(codeql, bqrsFilePath);
-  const compatibleQueryKinds = bqrsInfo.compatibleQueryKinds;
-  const queryMetadata = await getQueryMetadata(
+  // Calculate query run information like BQRS file paths, etc.
+  const queryPackRunResults = await getQueryPackRunResults(
     codeql,
-    await getRemoteQueryPackDefaultQuery(codeql, queryPack),
+    databaseName,
+    queryPaths,
+    queryPack,
+    queryPackName,
   );
 
   const sourceLocationPrefix = await getSourceLocationPrefix(codeql);
-  const sarifOutputType = getSarifOutputType(
-    queryMetadata,
-    compatibleQueryKinds,
+
+  const shouldGenerateSarif = await queryPackSupportsSarif(
+    codeql,
+    queryPackRunResults,
   );
+
   let resultCount: number;
   let sarifFilePath: string | undefined;
-  if (sarifOutputType !== undefined) {
+  if (shouldGenerateSarif) {
     const sarif = await generateSarif(
       codeql,
-      bqrsFilePath,
       nwo,
-      queryMetadata,
-      sarifOutputType,
       databaseName,
-      sourceLocationPrefix,
+      queryPackName,
       databaseSHA,
     );
     resultCount = getSarifResultCount(sarif);
     sarifFilePath = path.join("results", "results.sarif");
     fs.writeFileSync(sarifFilePath, JSON.stringify(sarif));
   } else {
-    resultCount = getBqrsResultCount(bqrsInfo);
+    resultCount = queryPackRunResults.totalResultsCount;
   }
 
+  const bqrsFilePaths = await adjustBqrsFiles(queryPackRunResults);
+
   return {
     resultCount,
     databaseSHA,
+    databaseName,
     sourceLocationPrefix,
-    bqrsFilePath,
+    bqrsFilePaths,
     sarifFilePath,
   };
 }
 
+async function adjustBqrsFiles(
```

Comment:
Do we need to move the physical BQRS files around on disk? When we create the results artifact we decide which file goes at which path in the zip.

Reply:
query.ts expects things to be in a `/results` dir so it seems a bit leaky to move that knowledge there.

Reply:
This repo is small enough that I think it's ok to have a slightly-more-complex interface between functions/files if it avoids us making the code more complex and having to do computational work to fit those interfaces.

Although, perhaps we can make the interface even more clear and also avoid copying the files around. Currently the unspoken assumption is that the BQRS files are contained in the `/results` dir, but how about if `RunQueryResult` included a `bqrsFilePathBase` (or whatever name) that indicates what all the BQRS files should be considered "relative to". That way `query.ts` doesn't have to assume it's the `/results` directory and we can keep the BQRS files where they are.

Reply:
Done!